WordPress/src/main/java/org/wordpress/android/ui/reader/discover/viewholders/ReaderPostNewViewHolder.kt

```diff
@@ -164,7 +167,14 @@ class ReaderPostNewViewHolder(
         view.isVisible = state.isEnabled
         view.isSelected = state.isSelected
         view.contentDescription = state.contentDescription?.let { uiHelpers.getTextOfUiString(view.context, it) }
-        view.setOnClickListener { state.onClicked?.invoke(postId, blogId, state.type) }
+        view.setOnClickListener {
+            // If it's a like action, we want to update the UI right away. If there's an error, we'll revert
+            // the UI change.
+            if (state.type == ReaderPostCardActionType.LIKE && networkUtilsWrapper.isNetworkAvailable()) {
```

Comment:
> networkUtilsWrapper.isNetworkAvailable()

This is ugly and shouldn't be here, but after spending a few hours trying to solve the problem the "right way" by using `PostLikeUseCase`, `ReaderPostCardActionsHandler` and a few other classes I realized it wasn't worth it. If you're aware of another way to handle the no internet connection scenario, please let me know.

Reply:
It appears that the UI is updated immediately when a network connection is detected, but independently of the network status, the events that attempt to contact the server are triggered. Therefore, if I'm on a spotty connection, there's a scenario where my first tap (like) doesn't update the UI, but the connection is established, and the server processes the like action for the post. Am I correct in assuming this behavior?

On the client side, is the UI updated when the server responds with success?

Reply:
After discussing the details on Slack, it appears that the scenario I mentioned in the comment above isn't even an edge case, because the networkUtilsWrapper.isNetworkAvailable() check would have returned true even on spotty connections.